File: YourQuoteBoard/YourQuoteBoard/DTO/Book/BookDisplayDTO.cs
﻿using YourQuoteBoard.DTO.Tag;

namespace YourQuoteBoard.DTO.Book
{
    public class BookDisplayDTO
    {
        public Guid BookId { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public string CoverImagePath { get; set; }
        public double? AverageRating { get; set; }
        public int? NumberOfRatings { get; set; }
        public ICollection<TagDisplayDTO> Tags { get; set; } = [];
    }
}

File: YourQuoteBoard.Client/src/models/books/BookDisplayDTO.ts
import { TagDisplayDTO } from "../tag/TagDisplayDTO";

export default interface BookDisplayDTO{
    bookId: string;
    title: string;
    author: string;
    coverImagePath: string;
    averageRating: number | null;
    numberOfRatings: number | null;
    tags: TagDisplayDTO[];
}